Dotcom Tests

Tonkpils opened the pull request:
This covers some of the basic tests that we spoke about
- Create Items 
- Create Purchase Orders
- Create Orders
- Get Order Status
- Get Inventory
- Get Returns

Note that some of these tests are failing because we don't have actual data to test with when making requests to the Sandbox API.

Tonkpils committed:
Dotcom remote tests.
Post item yells when quantity is exposed

Tonkpils committed:
Add tests for integration
Also fixes quantity on item when expected and proper purchase order errors

jimm commented on Rakefile:
```diff
@@ -19,6 +19,12 @@ namespace :test do
     t.libs << 'test'
     t.verbose = true
   end
+
```
How about adding this task's pattern to the `:dotcom` task above? Or is this temporary until you fix these tests?

jimm commented on lib/active_fulfillment/models/dotcom_distribution/post_item.rb:
```diff
@@ -118,7 +118,9 @@ def item_to_xml(xml)
             xml.cdata self.description
           end
-          xml.send(:"quantity", self.quantity, inject_nil(self.quantity))
+          if quantity
```
I don't see `quantity` in the docs, though I do see `package-qty`. Also, if it's absent shouldn't we use `inject_nil(self.quantity)` (or rather `package_qty`)?

Tonkpils replied:
`PurchaseOrder` uses `PostItem` to create a new purchase order for the specific item for a `quantity`. The catch is that PurchaseOrder requires item to have a quantity while PostItem will fail if quantity is present.

I think that the objects in the gem require some clean up as there's a bit of duplication going on. If anything, we could start thinking about removing ActiveModel and going with another approach but I think that's low priority.

Tonkpils committed:
Remove dotcom dist test task.
This task can be run with rake test:remote TEST=test/remote/dotcom_distribution_test.rb
